### PR TITLE
[frontend/backend] Create public dashboards from dashboard list pages (#6509)

### PR DESCRIPTION
**Pull request by lndrtrbn**
### Proposed changes

* Can create a new public dashboard from the page listing public dashboards
* Can create a new public dashboard from an item of the list of custom dashboards

### Related issues

* #6509 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

To test use at least 2 different users. For each one create several dashboards and give different access to the other user. When creating a public dashboard, pay attention to dashboards you can create a public and the ones you can't.

You should be able to create a new public dashboard only if you have "can manage" access.

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceShareForm.tsx
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/workspaces/WorkspaceShareForm.tsx:1]: Replaced by a new component `PublicDashboardCreationForm`

**Review by SouadHadjiat — APPROVED** [review]
tested locally ✅

**Comment by frapuks** [comment]
I have an issue, when a user have one dashbord (a a public dashboard associated), and an other user without dashboards click on `public dashboard`

FIRST USER : (create a dashboard and share it)
image

SECOND USER : (has no dashboard)
error

**Review by frapuks — APPROVED** [review]
tested locally ✅

**Pull request merged**